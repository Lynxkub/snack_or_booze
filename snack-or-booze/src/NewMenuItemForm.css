.input {
    float : right;
}

.radio {
   width : 300px;
   margin : auto;
   
}