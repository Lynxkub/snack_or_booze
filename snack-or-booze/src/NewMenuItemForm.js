import React , {useState} from 'react';
import {FormGroup} from 'reactstrap';
import createItem from './helpers';

const NewMenuItemForm = () => {
    const INITIAL_VALUE = {
        menu : '',
        itemName: '',
        recipe: '',
        serve: '',
        description: '',
        id : ''
    }
    const [formData , setFormData] = useState(INITIAL_VALUE);


    const handleChange = (e) => {
        const {name , value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name] : value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        createItem(formData);
        setFormData(INITIAL_VALUE);
    }
    return (
        <div className='radio'>
            <h2>Add a new snack or drink to our menu!</h2>
            <div>
        <form onSubmit = {handleSubmit}>
            <div className='radio'>
            <input type = 'radio' id='snacks' name='menu' value='snacks' onChange={handleChange} data-testid='snacks' />
            <label htmlFor='snacks' >Snack</label>
            <input type='radio' id='drinks' name='menu' value='drinks' onChange={handleChange} data-testid='drinks'  />
            <label htmlFor='drinks'>Drink</label>
            </div>
            <div>
            <label htmlFor='itemName'>Item Name</label>
            <input 
            id='itemName'
            type='text'
            name='itemName'
            placeholder='Item Name'
            value={formData.name}
            onChange={handleChange}
            className='input'
            data-testid='itemName'
            />
            </div>
            <div>
            <label htmlFor='description'>Description</label>
            <input 
            id='description'
            type='text'
            name='description'
            placeholder='Description'
            value={formData.description}
            onChange={handleChange}
            className='input'
            data-testid='description'
            />
            </div>
            <div>
            <label htmlFor='recipe'>Recipe</label>
            <input 
            id='recipe'
            type='text'
            name='recipe'
            placeholder='Recipe'
            value={formData.recipe}
            onChange={handleChange}
            className='input'
            data-testid='recipe'
            />
            </div>
            <div>
            <label htmlFor='serve'>Serving Instructions</label>
            <input 
            id='serve'
            type='text'
            name='serve'
            placeholder='Serving Instructions'
            value={formData.serve}
            onChange={handleChange}
            className='input'
            data-testid='serve'
            />
            </div>
            <button>Submit Item</button>
        </form>
        </div>
        </div>
    )
}


export default NewMenuItemForm;