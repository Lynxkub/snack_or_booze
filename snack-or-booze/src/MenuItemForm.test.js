import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NewMenuItemForm from './NewMenuItemForm';
import SnackOrBoozeApi from './Api';


it('renders without crashing' , () => {
    render(<NewMenuItemForm />)
})

it('matches snapshot' , () => {
    const {asFragment} = render(<NewMenuItemForm />);
    expect(asFragment()).toMatchSnapshot();
})

it('should add a new drink' , function () {
    const{queryByText , getByTestId} = render(<NewMenuItemForm />);
    const itemName = getByTestId('itemName')
    const drinks = getByTestId('drinks');
    const description = getByTestId('description');
    const recipe = getByTestId('recipe');
    const serve = getByTestId('serve');
    const btn = queryByText('Submit Item')
    expect(itemName).toBeInTheDocument();
   
    fireEvent.change(itemName , {target : {value : 'test item'}})
    fireEvent.click(drinks)
    fireEvent.change(description , {target : {value : 'test description'}})
    fireEvent.change(recipe , {target : {value : 'test recipe'}})
    fireEvent.change(serve , {target : {value : 'test serve'}})
    fireEvent.click(btn);

    setTimeout( async () => {
        const res = await SnackOrBoozeApi.getDrinks();
        expect(res[res.lenght]).toEqual({
            'id': 'test item',
            'name' : 'test item',
            'description' : 'test description',
            'recipe' : 'test recipe',
            'serve' : 'test serve'
        } , 2000)
    })
})
