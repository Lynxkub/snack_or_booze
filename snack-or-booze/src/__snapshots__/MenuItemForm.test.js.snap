// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="radio"
  >
    <h2>
      Add a new snack or drink to our menu!
    </h2>
    <div>
      <form>
        <div
          class="radio"
        >
          <input
            data-testid="snacks"
            id="snacks"
            name="menu"
            type="radio"
            value="snacks"
          />
          <label
            for="snacks"
          >
            Snack
          </label>
          <input
            data-testid="drinks"
            id="drinks"
            name="menu"
            type="radio"
            value="drinks"
          />
          <label
            for="drinks"
          >
            Drink
          </label>
        </div>
        <div>
          <label
            for="itemName"
          >
            Item Name
          </label>
          <input
            class="input"
            data-testid="itemName"
            id="itemName"
            name="itemName"
            placeholder="Item Name"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="description"
          >
            Description
          </label>
          <input
            class="input"
            data-testid="description"
            id="description"
            name="description"
            placeholder="Description"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="recipe"
          >
            Recipe
          </label>
          <input
            class="input"
            data-testid="recipe"
            id="recipe"
            name="recipe"
            placeholder="Recipe"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="serve"
          >
            Serving Instructions
          </label>
          <input
            class="input"
            data-testid="serve"
            id="serve"
            name="serve"
            placeholder="Serving Instructions"
            type="text"
            value=""
          />
        </div>
        <button>
          Submit Item
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
